### PR TITLE
Improve typing for ``mocksvgconverter.py``

### PR DESCRIPTION
**Pull request by adamtheturtle**
### Feature or Bugfix

- Refactoring

This is the new / better way of doing things, and is consistent with other parts of this codebase.

### CODE REVIEW

File: sphinx/util/display.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
 from __future__ import annotations
 
 import functools
+from typing import TYPE_CHECKING
```
Comment (AA-Turner) [inline sphinx/util/display.py]:
```suggestion
```

File: sphinx/util/display.py [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 
 from sphinx.locale import __
 from sphinx.util import logging
 from sphinx.util.console import bold, color_terminal
 
-if False:
+if TYPE_CHECKING:
```
Comment (AA-Turner) [inline sphinx/util/display.py]:
```suggestion
if False:
```

File: tests/roots/test-ext-imgmockconverter/mocksvgconverter.py
Comment (AA-Turner) [inline tests/roots/test-ext-imgmockconverter/mocksvgconverter.py:1]: Can we update the `# type` comments in this file, too?

**Review by AA-Turner — CHANGES_REQUESTED** [review]

File: tests/roots/test-ext-imgmockconverter/mocksvgconverter.py
Comment (adamtheturtle) [reply tests/roots/test-ext-imgmockconverter/mocksvgconverter.py:1]: Thanks for taking this on!

**Pull request merged**